src/core/hle/service/am/applets/web_browser.h

```diff
@@ -32,15 +39,39 @@ class WebBrowser final : public Applet {
     void Finalize();
 
 private:
+    void InitializeInternal();
+    void ExecuteInternal();
+
+    // Specific initializers for the types of web applets
+    void InitializeShop();
+    void InitializeOffline();
+
+    // Specific executors for the types of web applets
+    void ExecuteShop();
+    void ExecuteOffline();
+
     Core::Frontend::WebBrowserApplet& frontend;
 
+    // Extra frontends for specialized functions
+    Core::Frontend::ECommerceApplet* frontend_e_commerce;
+
     bool complete = false;
     bool unpacked = false;
     ResultCode status = RESULT_SUCCESS;
 
-    FileSys::VirtualFile manual_romfs;
+    ShimKind kind;
+    std::map<WebArgTLVType, std::vector<u8>> args;
+
+    FileSys::VirtualFile offline_romfs;
     std::string temporary_dir;
     std::string filename;
+
+    ShopWebTarget shop_web_target;
+    std::map<std::string, std::string> shop_query;
```

Comment:
This can be:

```suggestion
    std::map<std::string, std::string, std::less<>> shop_query;
```

to allow heterogenous lookup. This way lookups like:

```cpp
const auto scene = shop_query.find("scene");
```

won't cause scene to be converted into a std::string before doing the lookup.